packages/graphql-eslint-rules/src/missing-apollo-key-fields.ts

```diff
@@ -0,0 +1,263 @@
+    );
+  }
+
+  if (!context.parserServices.hasTypeInfo) {
+    throw new Error(
+      `Rule '${ruleName}' requires 'parserOptions.schema' to be set and schema to be loaded. See http://bit.ly/graphql-eslint-schema for more info`
+    );
+  }
+
+  if (!context.options[0]?.typePolicies) {
+    throw new Error(
+      `Rule '${ruleName}' requires option 'typePolicies' to be set.`
+    );
+  }
+}
+
+function keyFieldsForType(
+  type: GraphQLObjectType | GraphQLInterfaceType,
+  typePolicies: any
+) {
+  const keyFields: string[] = [];
+  const typePolicy = typePolicies[type.name];
+  if (typePolicy && typePolicy.keyFields) {
+    if (Array.isArray(typePolicy.keyFields)) {
+      typePolicy.keyFields.forEach((keyField: any) => {
+        if (typeof keyField === "string") {
+          keyFields.push(keyField);
+        } else {
+          throw new Error("Expected keyFields to be array of strings");
+        }
+      });
+    } else {
+      throw new Error("Expected keyFields to be array of strings");
+    }
+  } else if (type.getFields().id !== undefined) {
+    keyFields.push(DEFAULT_KEY_FIELD_NAME);
+  }
+  return keyFields;
+}
+
+function getKeyFieldsObjectForCheck(keyFields: string[]) {
+  return keyFields.reduce((acc, id) => {
+    acc[id] = false;
+    return acc;
+  }, {} as Record<string, boolean>);
+}
+
+function getUnusedKeyFields(idsObjectForCheck: Record<string, boolean>) {
+  return Object.entries(idsObjectForCheck).reduce((acc, [key, value]) => {
+    if (value) {
+      return acc;
+    }
+    acc.push(key);
+    return acc;
+  }, [] as string[]);
+}
+
+export default {
+  meta: {
+    type: "suggestion",
+    docs: {
+      category: "Best Practices",
+      description: `Enforce selecting specific fields when they are available on the GraphQL type.`,
+      requiresSchema: true,
+      requiresSiblings: true,
+      examples: [
+        {
+          title: "Incorrect",
+          code: /* GraphQL */ `
+            # In your schema
+            type User {
+              id: ID!
+              name: String!
+            }
+            # Query
+            query user {
+              user {
+                name
+              }
+            }
+          `,
+        },
+        {
+          title: "Correct",
+          code: /* GraphQL */ `
+            # In your schema
+            type User {
+              id: ID!
+              name: String!
+            }
+            # Query
+            query user {
+              user {
+                id
+                name
+              }
+            }
+          `,
+        },
+      ],
+    },
+    messages: {
+      [REQUIRE_KEY_FIELDS_WHEN_AVAILABLE]: `Field "{{ fieldName }}" must be selected when it's available on a type. Please make sure to include it in your selection set!\nIf you are using fragments, make sure that all used fragments {{checkedFragments}} specifies the field "{{ fieldName }}".`,
```

Comment:
What does the output look like when there are multiple offences in a file?

Reply:
Changed to ```Field(s) "{{ fieldName }}" must be selected (when available on a type). Please make sure to include it in your selection set!                                                                                                     
  If you are using fragments, make sure that all used fragments {{checkedFragments}} specifies the field(s) "{{ fieldName }}".```

it returns founded unusedKeyFields 
```               
data: {
       checkedFragments: checkedFragmentSpreads.size === 0 ? "" : `(${Array.from(checkedFragmentSpreads).join(", ")})`,
       fieldName: unusedKeyFields.join(", ")
}
```